.github/workflows/deploy.yml

```diff
@@ -0,0 +1,127 @@
+      ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      AWS_DEFAULT_REGION: ${{secrets.AWS_DEFAULT_REGION}}
+      SECRETS_S3_BUCKET: ${{ secrets.SECRETS_S3_BUCKET }}
+      SLACK_NOTIFICATION_CI_RUNS: ${{ secrets.SLACK_NOTIFICATION_CI_RUNS }}
+      REPOSITORY_URL: ${{secrets.ACCOUNT_ID}}.dkr.ecr.${{secrets.AWS_DEFAULT_REGION}}.amazonaws.com
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+      - name: Configuration
+        env:
+          AWS_CLI_VERSION: 1.18.101
+          DOCKER_VERSION: 20.10.3-r0
+          ECR_CREDENTIAL_HELPER_VERSION: 0.5.0
+        run: |
+          apk update 
+          apk add --no-cache curl bash coreutils skopeo jq py3-pip rsync git openssh openssh-client tzdata 
+          ln -s /usr/share/zoneinfo/UTC /etc/localtime 
+          apk add --no-cache docker==${DOCKER_VERSION} 
+          pip3 install awscli==${AWS_CLI_VERSION} 
+
+          curl -Lo /usr/local/bin/ecs-cli https://amazon-ecs-cli.s3.amazonaws.com/ecs-cli-linux-amd64-latest 
+          chmod +x /usr/local/bin/ecs-cli
+
+          curl -Lo /bin/docker-credential-ecr-login https://amazon-ecr-credential-helper-releases.s3.us-east-2.amazonaws.com/${ECR_CREDENTIAL_HELPER_VERSION}/linux-amd64/docker-credential-ecr-login 
+          chmod +x /bin/docker-credential-ecr-login
+
+          curl -o /tmp/terraform.zip https://releases.hashicorp.com/terraform/1.1.7/terraform_1.1.7_linux_amd64.zip 
+          unzip /tmp/terraform.zip -d /usr/bin
+      - name: Start deployment slack notification
+        run: |
+          curl -X POST -H 'Content-type: application/json' \
+          --data "{\"text\":\"The $SERVICE service(s) deployment on $ENV environment has begun.\n \"}" \
+          $SLACK_NOTIFICATION_CI_RUNS
+      - name: Set DOCKER_TAG
+        run: |
+          echo "DOCKER_TAG=${REPOSITORY_URL}/emrit-${ENV}-${SERVICE}:${BUILD_NUMBER}" >> $GITHUB_ENV
+      - name: Build image
+        run: docker build --compress -t ${DOCKER_TAG} --build-arg git_pat=${ACCESS_TOKEN} -f Dockerfile .
+      - name: docker login
+        run: docker login --username AWS --password "$(aws ecr get-login-password)" ${REPOSITORY_URL}
+      - name: push image
+        run: docker push ${DOCKER_TAG}
+      - name: Deploy to ECS
+        run: |
+          git clone --single-branch --branch ${INFRA_REPO_BRANCH} "https://${INFRA_ACCESS_TOKEN}@${INFRA_REPO}" infra
+          cd infra/env/${ENV}
+          aws s3 cp s3://${SECRETS_S3_BUCKET}/ . --recursive --exclude "*" --include "*.secret.auto.tfvars"
+
+          TF_IMAGES_FILE=${TF_IMAGES_FILE:-"images.secret.auto.tfvars"}
+          FIELDS=`echo $TF_IMAGE_FIELD | tr -s ',' ' '`
+          for field in $FIELDS; do
+              sed -i /${field}/d "${TF_IMAGES_FILE}"
+              echo "${field} = \"${BUILD_NUMBER}\"" >> "${TF_IMAGES_FILE}"
+          done
```

Comment:
Correct me if I'm wrong, but I think the point is replace a single line. For instance `search_image_version = "10"` with `search_image_version = "11"`. The simplest approach would be to delete the line matching the criteria and appending a new replacement line (this is the purpose of the snippet outlined [here](https://github.com/emritio/search/pull/2#discussion_r882523638)).

Reply:
Yeah so that's what we do within the for loop. The for loop allows us to have a comma or space separated list of values that we would replace at the same time. So if two modules were being updated at the same time to the same build number for example. Thinking about it though, we don't have a mechanism to build and deploy more than one image, so I don't know how we'd be in a position to do this anyway.... I had left it in because that's what they had before, but I think it's a case of over engineering something before it was needed